src/proc_fault/proc_fault_node.cc

```diff
@@ -45,25 +52,143 @@ namespace proc_fault
+            //mappingSoftwareInterface.push_back(new ProcMapping());
+        }
+
+        procFaultModule.push_back(new Module("Mapping", mappingSoftwareInterface));
+    }
+
+    void ProcFaultNode::initHydro()
+    {
+        std::vector<SoftwareInterface*> hydroSoftwareInterface;
+
+        if(Configuration::getInstance()->providerHydroEnable)
+        {
+            hydroSoftwareInterface.push_back(new ProviderHydrophone());
+        }
+
+        if(Configuration::getInstance()->procHydroEnable)
+        {
+            hydroSoftwareInterface.push_back(new ProcHydrophone());
+        }
+
+        procFaultModule.push_back(new Module("Hydro", hydroSoftwareInterface));
+    }
+
+    void ProcFaultNode::initIo()
+    {
+        std::vector<SoftwareInterface*> ioSoftwareInterface;
+
+        if(Configuration::getInstance()->providerActuatorEnable)
+        {
+            ioSoftwareInterface.push_back(new ProviderActuator());
+        }
+
+        if(Configuration::getInstance()->procActuatorEnable)
+        {
+            ioSoftwareInterface.push_back(new ProcActuator());
+        }
+
+        procFaultModule.push_back(new Module("Io", ioSoftwareInterface));
+    }
+
+    void ProcFaultNode::initUnderwaterCom()
+    {
+        std::vector<SoftwareInterface*> underwaterComSoftwareInterface;
+
+        if(Configuration::getInstance()->providerUnderwaterComEnable)
+        {
+            underwaterComSoftwareInterface.push_back(new ProviderCom());
+        }
         
-        Module* navigationModule = new Module("Navigation", navigationSoftwareInterface);
+        procFaultModule.push_back(new Module("Underwater Com", underwaterComSoftwareInterface));
+    }
+
+    void ProcFaultNode::initPower()
+    {
+        std::vector<SoftwareInterface*> powerSoftwareInterface;
+
+        if(Configuration::getInstance()->powerEnable)
+        {
+            powerSoftwareInterface.push_back(new ProviderPower());
+        }
+
+        if(Configuration::getInstance()->providerThrusterEnable)
+        {
+            powerSoftwareInterface.push_back(new ProviderThruster());
+        }
+        
+        procFaultModule.push_back(new Module("Power", powerSoftwareInterface));
+    }
+
+    void ProcFaultNode::initInternalCom()
+    {
+        std::vector<SoftwareInterface*> internalComSoftwareInterface;
 
-        procFaultModule.push_back(navigationModule);
+        if(Configuration::getInstance()->interfaceEnable)
+        {
+            internalComSoftwareInterface.push_back(new InterfaceRs485());
+        }
+        
+        procFaultModule.push_back(new Module("Internal Com", internalComSoftwareInterface));
     }
 
     void ProcFaultNode::spin()
     {
-        ros::Rate r(20);
+        ros::Rate r(5);
         while(ros::ok())
         {
             ros::spinOnce();
 
             sonia_common::FaultDetection msg;
             msg.navigation = procFaultModule[0]->checkMonitoring();
+            msg.vision = procFaultModule[1]->checkMonitoring();
```

Comment:
Can you make a loop of it?

Reply:
@supertoto29 I don't really understand, it's in the spin it should loop...